src/bdii/htcondor-ce-provider

```diff
@@ -0,0 +1,245 @@
+GLUE2EndpointServiceForeignKey: %(central_manager)s
+GLUE2ComputingEndpointComputingServiceForeignKey: %(central_manager)s
+"""
+share_ldif = """dn: GLUE2ShareID=%(shareid)s,%(bind_dn)s
+objectClass: GLUE2Entity
+objectClass: GLUE2Share
+objectClass: GLUE2ComputingShare
+GLUE2ShareID: %(shareid)s
+GLUE2ComputingShareServingState: production
+GLUE2ComputingShareTotalJobs: %(total_vo_jobs)s
+GLUE2ComputingShareWaitingJobs: %(idle_vo_jobs)s
+GLUE2ComputingShareRunningJobs: %(running_vo_jobs)s
+GLUE2ComputingShareComputingServiceForeignKey: %(central_manager)s
+GLUE2ComputingShareComputingEndpointForeignKey: %(endpointid)s
+GLUE2ComputingShareExecutionEnvironmentForeignKey: %(central_manager)s_standard
+GLUE2ShareServiceForeignKey: %(central_manager)s
+GLUE2ShareEndpointForeignKey: %(endpointid)s
+GLUE2ShareResourceForeignKey: %(central_manager)s_standard
+"""
+
+policy_ldif = """dn: GLUE2PolicyID=%(policyid)s,GLUE2ShareID=%(shareid)s,%(bind_dn)s
+objectClass: GLUE2Entity
+objectClass: GLUE2Policy
+objectClass: GLUE2MappingPolicy
+GLUE2PolicyID: %(policyid)s
+GLUE2PolicyScheme: org.glite.standard
+GLUE2PolicyRule: vo:%(vo)s
+GLUE2MappingPolicyShareForeignKey: %(shareid)s
+"""
+
+def main():
+    
+    # Get central_manager hostname
+    central_manager = htcondor.param.get('CENTRAL_MANAGER')
+    if not central_manager:
+        sys.stderr.write("Error: CENTRAL_MANAGER not set\n")
+        sys.exit(1)
+
+    # Get VO Names
+    vonames = htcondor.param.get('HTCONDORCE_VONames')
+    if not vonames:
+        sys.stderr.write("Error: HTCONDORCE_VONames not set\n")
+        sys.exit(1)
+    vonames = vonames.split(',')
+
+    # Get Site Name
+    site_name = htcondor.param.get('GLUE2DomainID')
+    if not site_name:
+        sys.stderr.write("Error: GLUE2DomainID: not set\n")
+        sys.exit(1)
+
+    # This is the bind DN for all entries
+    bind_dn = "GLUE2ServiceID=%s,GLUE2GroupID=resource,o=glue" % (central_manager)
+
+    # Query collector for number of CPUs and Collector ad
+    coll = htcondor.Collector(central_manager)
+    total_cores = 0
+    for ad in coll.query(htcondor.AdTypes.Startd, 'State=!="Owner"', ['Cpus']):
+        if ad.get('Cpus'):
+            total_cores += ad['Cpus']
+    coll_ad = coll.locate(htcondor.DaemonTypes.Collector)
+    version = coll_ad['CondorVersion'].split()[1]
+
+    # Print the entry for the GLUE2 Manager
+    print manager_ldif %{
+        'central_manager': central_manager, 
+        'bind_dn': bind_dn, 
+        'version': version,
+        'total_cores': total_cores,
+        }
+
+    instances = coll_ad['HostsUnclaimed']
+    # Print the entry for the GLUE2 Resource
+    print resource_ldif %{
+        'central_manager': central_manager, 
+        'bind_dn': bind_dn,
+        'instances': instances,
+        }
+
+    schedd_ads = coll.locateAll(htcondor.DaemonTypes.Schedd)
+
+    # Print the entry for the GLUE2 Service
+    print service_ldif %{
```

Comment:
Since Python 2 is pretty much going into legacy mode, wouldn't it make sense to `from __future__ import print_function` and use `"{name}".format(name=value)` instead of `"%(name)" % {name:value}`?